api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -220,6 +232,89 @@ type AzureManagedControlPlaneSpec struct {
 	// DisableLocalAccounts disables getting static credentials for this cluster when set. Expected to only be used for AAD clusters.
 	// +optional
 	DisableLocalAccounts *bool `json:"disableLocalAccounts,omitempty"`
+
+	// SecurityProfile defines the security profile for the container service cluster.
+	// +optional
+	SecurityProfile *ManagedClusterSecurityProfile `json:"securityProfile,omitempty"`
+}
+
+// ManagedClusterSecurityProfile defines the security profile for the container service cluster.
+type ManagedClusterSecurityProfile struct {
+	// AzureKeyVaultKms defines Azure Key Vault key management service settings for the security profile.
+	// +optional
+	AzureKeyVaultKms *AzureKeyVaultKms `json:"azureKeyVaultKms,omitempty"`
+
+	// Defender defines Microsoft Defender settings for the security profile.
+	// +optional
+	Defender *ManagedClusterSecurityProfileDefender `json:"defender,omitempty"`
+
+	// ImageCleaner settings for the security profile.
+	// +optional
+	ImageCleaner *ManagedClusterSecurityProfileImageCleaner `json:"imageCleaner,omitempty"`
+
+	// Workloadidentity enables Kubernetes applications to access Azure cloud resources securely with Azure AD. Ensure to enable OIDC issuer while enabling Workload Identity
+	// +optional
+	WorkloadIdentity *ManagedClusterSecurityProfileWorkloadIdentity `json:"workloadIdentity,omitempty"`
```

Comment:
Why is the type a struct here? Can't we simply use bool for `WorkloadIdentity`? The struct anyways just have a bool field?
Are there any other configurables/fields that are possible for  `ManagedClusterSecurityProfileWorkloadIdentity`?